tools/checkpoint_conversion/checkpoint_conversion_utils.py

```diff
@@ -20,3 +22,12 @@ def get_md5_checksum(file_path):
         for byte_block in iter(lambda: f.read(4096), b""):
             md5_hash.update(byte_block)
     return md5_hash.hexdigest()
+
+
+def file_type_extractor(archive_file_path):
+    if archive_file_path.endswith(".tar.gz"):
+        with tarfile.open(archive_file_path, "r:gz") as tar:
+            return tar.extractall()
+    elif archive_file_path.endswith(".zip"):
+        with zipfile.ZipFile(archive_file_path, "r") as zip_ref:
+            return zip_ref.extractall()
```

Comment:
One question I had was: is there a way to specify the destination folder? Maybe, we should have that too?

Reply:
Yes! We can specify the destination folder inside `zip_ref/tar.extractall()`. I was initially doing it by passing `extract_dir` inside the `exctract_all()` method but it kept showing an error of `File not found`. If we do specify the destination folder, then we would have to change the path at other places as well I guess.

Reply:
cc: @abheesht17